### PR TITLE
Make the specs pass using pg adapter

### PR DESCRIPTION
**Pull request by erik-escobedo**
Before going anywhere else, it seems necessary to make the code work with a `pg` adapter, rather than `sqlite3`. This way, we could be able to run actual tests on our existing database, which I believe will turn out to be very useful :)

This changes introduce no rubocop complains or coverage loss

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -1,3 +1,6 @@
 test:
-  database: ':memory:'
-  adapter: sqlite3
+  adapter: postgresql
+  host: localhost
+  username: toptal
+  password: toptal
+  database: export
```
Comment (gvieira) [inline config/database.yml]: I suggest you configure your editor to always add a final line on saving. And do it here, as well.

File: lib/export/model.rb [hunk 2]
```diff
@@ -289,7 +289,7 @@ def hard_scope
           binds = nil
           node = nil
 
-          dependency.models.each do |model|
+          dependency.models.sort_by { |model| model.clazz.name }.each do |model|
```
Comment (gvieira) [inline lib/export/model.rb]: Why was this needed?

File: spec/support/statement_matcher.rb [hunk 3]
```diff
@@ -24,7 +24,10 @@
         raise 'expected that binds would match'
       end
 
-      ActiveRecord::Base.connection.raw_connection.prepare(actual_sql).close
+      ActiveRecord::Base.connection.raw_connection.tap do |connection|
```
Comment (gvieira) [inline spec/support/statement_matcher.rb]: Awesome! 👍

File: config/database.yml (hunk 1 above)
Comment (erik-escobedo) [reply config/database.yml]: I'm sorry. New computer. I'll configure it so right now

File: lib/export/model.rb (hunk 2 above)
Comment (erik-escobedo) [reply lib/export/model.rb]: Postgres adapter seems to always use alphabetical order in all of its operations. I don't fully understand why, but `circular_dependencies` method was returning different values for `sqlite3` and `pg` adapters until I added that sorting

File: spec/support/statement_matcher.rb (hunk 3 above)
Comment (erik-escobedo) [reply spec/support/statement_matcher.rb]: Thanks! I think `ActiveRecord::Base.connection.raw_connection.tap do |pg|` may represent intentions better. What do you think?

Comment (erik-escobedo) [reply spec/support/statement_matcher.rb]: I mean, `exec('DEALLOCATE . . .')` seems like a `pg` thing

File: lib/export/model.rb (hunk 2 above)
Comment (gvieira) [reply lib/export/model.rb]: OK then. Maybe you could sort the models when they are created. Or even better, use `ORDER BY` in the query that finds out the models for polymorphic relationships (in `dependency.rb`).

File: spec/support/statement_matcher.rb (hunk 3 above)
Comment (gvieira) [reply spec/support/statement_matcher.rb]: No problem by me.

File: lib/export/model.rb (hunk 2 above)
Comment (erik-escobedo) [reply lib/export/model.rb]: Great idea! Let me update the PR

File: config/database.yml [hunk 4]
```diff
@@ -1,3 +1,5 @@
 test:
-  database: ':memory:'
-  adapter: sqlite3
+  adapter: postgresql
+  host: localhost
+  username: toptal
```
Comment (jonatas) [inline config/database.yml]: Can we drop this at all and add a simple `config/database.yml.sample`?

File: .gitignore [hunk 5]
```diff
@@ -10,3 +10,4 @@
 .DS_Store
 .byebug_history
 coverage
+/config/database.yml
```
Comment (jonatas) [inline .gitignore]: Is this `/` in the start effective? It's just `config/database.yml` I think.
